src/Monolog/Logger.php

```diff
@@ -362,7 +365,7 @@ public function addRecord($level, $message, array $context = array())
             'context' => $context,
             'level' => $level,
             'level_name' => $levelName,
-            'channel' => $this->name,
+            'channel' => $sourceChannel !== null ? $sourceChannel : $this->name,
```

Comment:
Instead of doing this here, can you do the following at the top of the script . . .

`$sourceChannel ?: $this->name`

and then just use `$sourceChannel` here and on line 347?